source/math/FloatTransform3D.ooc

```diff
@@ -36,65 +37,81 @@ FloatTransform3D: cover {
 					case 0 => result = this a
 					case 1 => result = this b
 					case 2 => result = this c
-					case => OutOfBoundsException new(y, 3) throw()
+					case 3 => result = this d
+					case => OutOfBoundsException new(y, 4) throw()
 				}
 			case 1 =>
 				match (y) {
-					case 0 => result = this d
-					case 1 => result = this e
-					case 2 => result = this f
-					case => OutOfBoundsException new(y, 3) throw()
+					case 0 => result = this e
+					case 1 => result = this f
+					case 2 => result = this g
+					case 3 => result = this h
+					case => OutOfBoundsException new(y, 4) throw()
 				}
 			case 2 =>
 				match (y) {
-					case 0 => result = this g
-					case 1 => result = this h
-					case 2 => result = this i
-					case => OutOfBoundsException new(y, 3) throw()
+					case 0 => result = this i
+					case 1 => result = this j
+					case 2 => result = this k
+					case 3 => result = this l
+					case => OutOfBoundsException new(y, 4) throw()
 				}
 			case 3 =>
 				match (y) {
-					case 0 => result = this j
-					case 1 => result = this k
-					case 2 => result = this l
-					case => OutOfBoundsException new(y, 3) throw()
+					case 0 => result = this m
+					case 1 => result = this n
+					case 2 => result = this o
+					case 3 => result = this p
+					case => OutOfBoundsException new(y, 4) throw()
 				}
 			case => OutOfBoundsException new(x, 4) throw()
 		}
 		result
 	}
-	determinant ::= this a * (this e * this i - this f * this h) + this d * (this h * this c - this i * this b) + this g * (this b * this f - this e * this c)
-	translation ::= FloatSize3D new(this j, this k, this l)
+	determinant: Float {
+		get {
+			this a * this f * this k * this p + this a * this j * this o * this h + this a * this n * this g * this l +
+			this e * this b * this o * this l + this e * this j * this c * this p + this e * this n * this k * this d +
+			this i * this b * this g * this p + this i * this f * this o * this d + this i * this n * this c * this h +
+			this m * this b * this k * this h + this m * this f * this c * this l + this m * this j * this g * this d -
+			this a * this f * this o * this l - this a * this j * this g * this p - this a * this n * this k * this h -
+			this e * this b * this k * this p - this e * this j * this o * this d - this e * this n * this c * this l -
+			this i * this b * this o * this h - this i * this f * this c * this p - this i * this n * this g * this d -
+			this m * this b * this g * this l - this m * this f * this k * this d - this m * this j * this c * this h
+		}
+	}
```

Comment:
I won't trust this to work until there's a test for it. Try this:
`sudo apt-get install octave`
the launch `octave`
type `m = rand(4,4)` (or make your own)
then `det(m)`